fluentd/configs.d/filter/k8s_record_transform.conf

```diff
@@ -1,4 +1,24 @@
-<filter kubernetes.**>
+<filter kubernetes.journal.container**>
+  type record_transformer
+  enable_ruby
+  <record>
+    hostname ${(kubernetes_host rescue nil) || File.open('/etc/docker-hostname') { |f| f.readline }.rstrip}
+    # log is set if MESSAGE is JSON valued and has err or msg field
+    message ${record['MESSAGE'] || log}
+    version 1.2.0
+    time ${Time.at((record["_SOURCE_REALTIME_TIMESTAMP"] || record["__REALTIME_TIMESTAMP"]).to_f / 1000000.0).to_datetime.rfc3339(6)}
```

Comment:
It is different because log records that come from a kubernetes container will have the kubernetes_host field added by the k8s metadata plugin.  For syslog records, there is only the _HOSTNAME field.